roofit/roofitcore/inc/RooSpan.h

```diff
@@ -117,9 +117,16 @@ class RooSpan {
     return _span.data();
   }
 
+#ifdef NDEBUG
   constexpr typename std::span<T>::reference operator[](typename std::span<T>::index_type i) const noexcept {
     return _span[i];
   }
+#else
+  typename std::span<T>::reference operator[](typename std::span<T>::index_type i) const noexcept {
+    assert(i < _span.size());
```

Comment:
R__ASSERT?

Also how evil is it to change constexprness of a method depending on the build optimization level? (potentially a lot?)

Reply:
Axel says that `constexpr` is completely irrelevant here, so I can just remove it for good. This thing can get inlined everywhere, so it doesn't really matter how it's annotated.

But tell me this:
Why `R__ASSERT`? It's never used in RooFit, so I don't know why it would be better than the good old assert.

Reply:
I _think_ R__ASSERT's behavior can be customized by frameworks via error handlers, but here I was just suggesting it for consistency of the printouts. Which of course does not apply if roofit does not use R__ASSERT.